### PR TITLE
HZN-81: Documentation for Service Assurance

### PR DESCRIPTION
**Pull request by indigo423**
Documentation for OpenNMS users to explain core concepts of service assurance. Covering availability and measuring service response times. Explain how services, IP interfaces and nodes are represented. Which components interact with Pollerd and how service monitors are defined. User should know which Events Pollerd generates to understand what causes this events. Explanation of node correlation based on critical services and explain Path Outage behavior.

Create OpenNMS administrator documentation for service assurance with Pollerd. Covering configuration for Pollerd, Path Outage, Critical Service, reload configuration and Polling packages.

JIRA: http://issues.opennms.org/browse/HZN-81
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS288

### CODE REVIEW

File: opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+This section will cover the basic functionalities how _OpenNMS_ monitors availability and latency from applications or management agents.
+To change the behavior of how _OpenNMS_ monitors applications or status information from management agents please see _Administration Guide_.
+Extending the _Service Monitor_ framework please see _Development Guide_ of the _OpenNMS_ documentation.
```
Comment (StCyr) [inline opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc]: *To extend* the _Service Monitor_ framework,...

File: opennms-doc/guide-user/src/asciidoc/text/service-assurance/path-outage.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+For this reason it is possible to define a _Critical Path_.
+The _Critical Path_ has to be defined from the network perspective of the monitoring system.
+The following image shows a simple example how devices can be reached in a Layer 3 network topology.
+
+[[path-outage-example]]
+.Path Outage example
+image::service-assurance/04_path-outage.png[width=600]
+
+In figure <<path-outage-example, Path Outage example>> implies a network Layer 3 topology.
+From perspective of the monitoring system, _default-gw-01_ is on the _Critical Path_ to two networks.
```
Comment (StCyr) [inline opennms-doc/guide-user/src/asciidoc/text/service-assurance/path-outage.adoc]: From *the* perspective of the monitoring system,...

**Comment by indigo423** [comment]
:+1: @StCyr

File: opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+This section will cover the basic functionalities how _OpenNMS_ monitors availability and latency from applications or management agents.
+To change the behavior of how _OpenNMS_ monitors applications or status information from management agents please see _Administration Guide_.
```
Comment (j-white) [inline opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc]: see *the* Administration Guide

File: opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+To extend the _Service Monitor_ framework please see _Development Guide_ of the _OpenNMS_ documentation.
```
Comment (j-white) [inline opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc]: see *the* Development Guide

'of the OpenNMS documentation' can be omitted here.

File: opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+Measuring availability and latency of network services or applications is an important part in fault- and performance management.
```
Comment (j-white) [inline opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc]: 'fault-' vs fault

File: opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+| `uei.opennms.org/nodes/nodeLostService`     | _Critical Services_ are still up, just this service is lost.
+| `uei.opennms.org/nodes/nodeRegainedService` | Service came back up
+| `uei.opennms.org/nodes/interfaceDown`       | _Critical Service_ on an IP interface is down or all services are down.
+| `uei.opennms.org/nodes/interfaceUp`         | _Critical Service_ on that interface came back up again
+| `uei.opennms.org/nodes/nodeDown`            | All critical services on all IP interfaces are down from node.
+                                                The whole host is unreachable over the network.
+| `uei.opennms.org/nodes/nodeUp`              | Some of the _Critical Services_ came back online.
+|===
+
+The behavior to generate _interfaceDown_ and _nodeDown_ events is described in the <<gu-service-assurance-critical-service, Critical Service>> section.
```
Comment (j-white) [inline opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc]: This assumes that node-outage processing is enabled. Should we make a note of that here? Or is it out of scope of the user guide.

File: opennms-doc/guide-user/src/asciidoc/text/service-assurance/critical-service.adoc [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+[[gu-service-assurance-critical-service]]
+=== Critical service
+
+Monitoring services on an _IP network_ is resource expensive, especially if you have a lot of services to monitor which are not available.
+In case of an outage, the monitoring system uses most of its time waiting for retries and timeouts.
+To be more efficient, _OpenNMS_ uses by default _ICMP_ as a critical service.
+If the IP interface for example doesn't reply on layer 3 _ICMP_ echo requests, all other services will be not tested and are assumed as _Down_.
```
Comment (j-white) [inline opennms-doc/guide-user/src/asciidoc/text/service-assurance/critical-service.adoc]: I would rephrase these lines with something like:
```
Monitoring services on an _IP network_ can be resource expensive, especially in cases where many of these services are not available.
When a service is offline, or unreachable, the monitoring system spends most of it's time waiting for retries and timeouts.

In order to improve efficiency, _OpenNMS_ deems all services on a interface to be _Down_ if the critical service is _Down_.
By default _OpenNMS_ uses _ICMP_ as the critical service.
```

**Comment by j-white** [comment]
Looks great overall. I've added a few comments inline.

File: opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc (hunk 6 above)
Comment (indigo423) [inline opennms-doc/guide-user/src/asciidoc/text/service-assurance/introduction.adoc]: Yes, make sense, added a note here.

**Comment by indigo423** [comment]
@j-white perfect. Thank you very much for improvement. Worked all in.

**Comment by indigo423** [comment]
@j-white Do you have any preferences for target branch? Do you think it make sense to push it to foundation / foundation-2016 instead of develop to avoid back porting?

**Comment by j-white** [comment]
I don't think we should bother back-porting documentation in general, unless we're fixing inaccuracies.
